sdk/route/src/manual.ts

```diff
@@ -125,24 +130,55 @@ export class NttManualRoute<N extends Network>
     return { valid: true, params: validatedParams };
   }
 
-  async quote(params: Vp): Promise<QR> {
-    return {
+  async quote(
+    request: routes.RouteTransferRequest<N>,
+    params: Vp
+  ): Promise<QR> {
+    const result: QR = {
       success: true,
       params,
       sourceToken: {
-        token: this.request.source.id,
+        token: request.source.id,
         amount: params.normalizedParams.amount,
       },
       destinationToken: {
-        token: this.request.destination.id,
-        amount: amount.parse(params.amount, this.request.destination.decimals),
+        token: request.destination.id,
+        amount: amount.parse(params.amount, request.destination.decimals),
       },
     };
+    const { fromChain, toChain } = request;
+    const dstNtt = await toChain.getProtocol("Ntt", {
+      ntt: params.normalizedParams.destinationContracts,
+    });
+    const duration = await dstNtt.getRateLimitDuration();
+    if (duration > 0n) {
+      const capacity = await dstNtt.getCurrentInboundCapacity(fromChain.chain);
+      const dstAmount = amount.parse(
+        params.amount,
+        request.destination.decimals
+      );
+      if (
+        NttRoute.isCapacityThresholdExceeded(amount.units(dstAmount), capacity)
+      ) {
+        result.warnings = [
+          {
+            type: "DestinationCapacityWarning",
```

Comment:
extremely minor comment, this could be a shared const. but im fine with it the way it is

Reply:
agreed, will revisit when we add more warnings. don't want to make changes now to dismiss approvals